packages/basti-cdk/src/basti-instance.ts

```diff
@@ -0,0 +1,183 @@
+import { Construct } from 'constructs';
+import { aws_ec2, aws_iam, Tags } from 'aws-cdk-lib';
+
+import { BASTION_INSTANCE_CLOUD_INIT } from './bastion-cloudinit';
+import { generateShortId } from './basti-helper';
+import {
+  BASTION_INSTANCE_CREATED_BY_TAG_NAME,
+  BASTION_INSTANCE_ID_TAG_NAME,
+  BASTION_INSTANCE_IN_USE_TAG_NAME,
+  BASTION_INSTANCE_NAME_PREFIX,
+  BASTION_INSTANCE_ROLE_NAME_PREFIX,
+  BASTION_INSTANCE_SECURITY_GROUP_NAME_PREFIX,
+} from './basti-constants';
+
+/**
+ * The machine configuration for the bastion instance.
+ */
+export interface BastiMachineProps {
+  /**
+   * The instance type to use for the bastion instance.
+   *
+   * @default t2.micro
+   */
+  readonly instanceType?: aws_ec2.InstanceType;
+  /**
+   * The machine image to use for the bastion instance.
+   */
+  readonly machineImage?: aws_ec2.IMachineImage;
+}
+
+/**
+ * The properties for the bastion instance.
+ */
+export interface BastiInstanceProps {
+  /**
+   * The VPC to deploy the bastion instance into.
+   */
+  readonly vpc: aws_ec2.IVpc;
+
+  /**
+   * (Optional) The ID of the bastion instance. The ID will be used to identify
+   * the bastion instance. If not specified, a random ID will be generated.
+   *
+   * @default - A 8-character pseudo-random string
+   */
+  readonly bastiId?: string;
+
+  /**
+   * (Optional) The subnet selection to deploy the bastion instance into.
+   * If not specified, the default subnet selection will be used.
+   *
+   * @default - Public subnets in the VPC
+   */
+  readonly subnetSelection?: aws_ec2.SubnetSelection;
+
+  /**
+   * (Optional) The machine configuration to use for the bastion instance.
+   * If not specified, the default configuration will be used.
+   */
+  readonly machineConfig?: BastiMachineProps;
```

Comment:
The reason I set it up like this is that sometimes these constructs take many arguments. Having everything top level makes it a lot more difficult to see which parameters are related to each other.

I can flatten it out, but I would prefer not too. Instead, I can add documentation as to what a `machineConfig` is. I think that it can have a strong definition, something similar to.

machineConfig is all the parameters that influence the underlying EC2 instance. This could be from block devices that should be attached or the size of current storage.